### ISSUE TITLE
Kubenet/Flannel networkPlugin deployments failing

### ISSUE THREAD

**Issue by marcosmarcusm** [Azure/aks-engine#444]
**Is this a request for help?**:
YES
---

**Is this an ISSUE or FEATURE REQUEST?** (choose one):
ISSUE
---

**What version of aks-engine?**:
0.29.1
---

**Kubernetes version**:
      "orchestratorRelease": "1.12",
      "orchestratorVersion": "1.12.4",

**What happened**:
The initial deployment failed with error:
13:12:40 -0800   **KubeletNotReady**              runtime network not ready: NetworkReady=false reason:NetworkPluginNotReady message:Network plugin returns error: cni config load failed: no network config found in **/etc/cni/net.d: cni plugin not initialized: failed to load cni config**

**What you expected to happen**:
I expect the deployment to be successful.

**How to reproduce it** (as minimally and precisely as possible):
Please reference the cluster-deifnition.json attached.
[cluster-definition.txt](https://github.com/Azure/aks-engine/files/2833467/cluster-definition.txt)

**Anything else we need to know**:
= When referencing documentation here: https://kubernetes.io/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins/ We confirm **/etc/cni/net.d** is the location for azure-cni specific network config and not kubenet. 

= When we try to deploy again by specifying **azure** in the networkPlugin property, we don't experience the issue. We also tried deploying with **flannel** networkPlugin and experienced the same issue. 

**It seems that kubelet config is still being set for Azure CNI?**

**Comment by CecileRobertMichon** [comment]
@marcosmarcusm can you please share your full cluster definition for repro? I can't see what "Please reference the cluster-deifnition.json included at beginning of issue." is referring to, I can only see your k8s version definition.

**Comment by marcosmarcusm** [comment]
I attached that now. Thanks for the prompt reply.

**Comment by CecileRobertMichon** [comment]
Can confirm with containerd + kubenet nodes are not becoming ready:

```
k8s-agentpool1-20908056-0   NotReady   agent     30m       v1.12.5   <none>        Ubuntu 16.04.5 LTS   4.15.0-1036-azure   containerd://1.1.5
k8s-agentpool1-20908056-1   NotReady   agent     30m       v1.12.5   <none>        Ubuntu 16.04.5 LTS   4.15.0-1036-azure   containerd://1.1.5
k8s-master-20908056-0       NotReady   master    30m       v1.12.5   <none>        Ubuntu 16.04.5 LTS   4.15.0-1036-azure   containerd://1.1.5
```

**Comment by CecileRobertMichon** [comment]
we have a working test for k8s 1.10 with flannel + kubenet

**Comment by siddray200** [comment]
Yes and if you run: kubectl get pods --all-namespaces you will find many containers in ContainerCreating and Pending state.

And when you run kubectl describe node <node-linux-vmss> you will find the reason as:
13:12:40 -0800   KubeletNotReady              runtime network not ready: NetworkReady=false reason:NetworkPluginNotReady message:Network plugin returns error: cni config load failed: no network config found in /etc/cni/net.d: cni plugin not initialized: failed to load cni config

**Comment by MarcosMMartinez** [comment]
@CecileRobertMichon Should we consider this issue acknowledged, i.e. there's a path to resolution?

@siddray200 I understand you've manually installed flannel on your end. Can you evaluate if deploying k8s 1.10.* will provide any blockers compared to 1.11.* or 1.12.*? I ask because we haven't confirmed if you'll be deploying new cluster relatively frequently.

**Comment by b333z** [comment]
Sounds like the missing toleration on node NotReady, I'm seeing the same with cilium.

If so, the flannel side has a fix merged from https://github.com/Azure/aks-engine/pull/313 and looks to be included in v0.30.0 here: https://github.com/Azure/aks-engine/commit/7809548dc75a7b6d4919f052448ddfec28fe6fe8

So might be worth a re-test with [v0.30.0](https://github.com/Azure/aks-engine/releases/tag/v0.30.0) if that helps.

**Comment by siddray200** [comment]
So, I tried aks-engine v0.31.1 and Flannel is working directly with the template file. This is fixed. Thanks a lot for all your effort.
